tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+                      </Stack.Item>
+                    </Stack>
+                    {item.tool_paths.map((item) => (
+                      <Box key={item}>{item}</Box>
+                    ))}
+                  </Box>
+                )}
+                {item.machinery && (
+                  <Box>
+                    <Stack my={1}>
+                      <Stack.Item grow>
+                        <hr style={{ 'border-color': '#111' }} />
+                      </Stack.Item>
+                      <Stack.Item color={'gray'}>Machinery</Stack.Item>
+                      <Stack.Item grow>
+                        <hr style={{ 'border-color': '#111' }} />
+                      </Stack.Item>
+                    </Stack>
+                    {item.machinery.map((item) => (
+                      <Box key={item}>{item}</Box>
+                    ))}
+                  </Box>
+                )}
+              </Box>
+              {item.is_guide && !!item.steps?.length && (
+                <Box>
+                  <Stack my={1}>
+                    <Stack.Item grow>
+                      <hr style={{ 'border-color': '#111' }} />
+                    </Stack.Item>
+                    <Stack.Item color={'gray'}>Steps</Stack.Item>
+                    <Stack.Item grow>
+                      <hr style={{ 'border-color': '#111' }} />
+                    </Stack.Item>
+                  </Stack>
+                  <ul style={{ 'padding-left': '20px' }}>
+                    {item.steps.map((step) => (
+                      <li key={step}>{step}</li>
+                    ))}
+                  </ul>
+                </Box>
+              )}
+            </Stack.Item>
+            <Stack.Item>
+              {!item.is_guide && (
+                <Button
+                  width="104px"
+                  lineHeight={2.5}
+                  align="center"
+                  content="Make"
+                  disabled={!props.craftable || props.busy}
+                  icon={props.busy ? 'circle-notch' : 'utensils'}
+                  iconSpin={props.busy ? 1 : 0}
+                  onClick={() =>
+                    act('make', {
+                      recipe: item.ref,
+                    })
+                  }
+                />
+              )}
+              {item.foodtypes?.length > 0 && (
+                <Box color={'gray'} width={'104px'} lineHeight={1.5} mt={1}>
+                  {item.foodtypes.map((foodtype) => (
+                    <TypeContent
+                      key={item.ref}
+                      type={foodtype}
+                      diet={props.diet}
+                    />
+                  ))}
+                </Box>
+              )}
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const IngredientContent = (props) => {
+  const i = props.item;
+  return (
+    i.path.includes('/obj/item/food/') ? (
```

Comment:
```suggestion
  const { i: { amount, path, name } = props;
  return (
   path?.includes('/obj/item/food/') ? (
```
destructure as much as possible!
also, what is "i"? can we make this not a one-letter var?